File: convert-pdf.py
#!/usr/bin/env python3

from weasyprint import HTML, CSS
import os
import sys

def convert_to_pdf(html_file, pdf_file):
    """Convert HTML file to PDF using WeasyPrint"""
    try:
        html_path = os.path.abspath(html_file)
        pdf_path = os.path.abspath(pdf_file)
        
        print(f"Converting {html_file}...")
        print(f"  Input:  {html_path}")
        print(f"  Output: {pdf_path}")
        
        HTML(html_path).write_pdf(pdf_path)
        
        # Check file size
        file_size = os.path.getsize(pdf_path) / (1024 * 1024)
        print(f"✓ {os.path.basename(pdf_file)} created ({file_size:.2f} MB)")
        return True
        
    except Exception as e:
        print(f"✗ Error converting {html_file}: {e}")
        return False

def main():
    base_dir = os.path.dirname(os.path.abspath(__file__))
    
    # Convert English manual
    en_html = os.path.join(base_dir, 'manual-en.html')
    en_pdf = os.path.join(base_dir, 'manual-en.pdf')
    
    success_en = convert_to_pdf(en_html, en_pdf)
    
    print()
    
    # Convert Filipino manual
    fil_html = os.path.join(base_dir, 'manual-fil.html')
    fil_pdf = os.path.join(base_dir, 'manual-fil.pdf')
    
    success_fil = convert_to_pdf(fil_html, fil_pdf)
    
    print()
    
    if success_en and success_fil:
        print("✅ Both PDFs completed successfully!")
        print(f"English: {en_pdf}")
        print(f"Filipino: {fil_pdf}")
        return 0
    else:
        print("❌ One or more conversions failed")
        return 1

if __name__ == '__main__':
    sys.exit(main())

File: convert-pdf.js
const puppeteer = require('puppeteer');
const path = require('path');
const fs = require('fs');

(async () => {
  let browser;
  try {
    browser = await puppeteer.launch({
      headless: 'new',
      args: [
        '--no-sandbox',
        '--disable-setuid-sandbox',
        '--disable-dev-shm-usage'
      ]
    });

    // Test with English manual
    console.log('Converting manual-en.html to PDF...');
    const enPage = await browser.newPage();
    
    const enPath = `file://${path.resolve(__dirname, 'manual-en.html')}`;
    console.log(`Loading: ${enPath}`);
    
    const response = await enPage.goto(enPath, {
      waitUntil: 'networkidle2',
      timeout: 90000
    });
    
    if (!response.ok) {
      throw new Error(`Failed to load page: ${response.status()}`);
    }
    
    console.log('Page loaded. Generating PDF...');
    
    await enPage.pdf({
      path: path.resolve(__dirname, 'manual-en.pdf'),
      format: 'A4',
      printBackground: true,
      margin: {
        top: 0,
        right: 0,
        bottom: 0,
        left: 0
      }
    });
    
    console.log('✓ manual-en.pdf created');
    await enPage.close();

    // Test with Filipino manual
    console.log('\nConverting manual-fil.html to PDF...');
    const filPage = await browser.newPage();
    
    const filPath = `file://${path.resolve(__dirname, 'manual-fil.html')}`;
    console.log(`Loading: ${filPath}`);
    
    const filResponse = await filPage.goto(filPath, {
      waitUntil: 'networkidle2',
      timeout: 90000
    });
    
    if (!filResponse.ok) {
      throw new Error(`Failed to load page: ${filResponse.status()}`);
    }
    
    console.log('Page loaded. Generating PDF...');
    
    await filPage.pdf({
      path: path.resolve(__dirname, 'manual-fil.pdf'),
      format: 'A4',
      printBackground: true,
      margin: {
        top: 0,
        right: 0,
        bottom: 0,
        left: 0
      }
    });
    
    console.log('✓ manual-fil.pdf created');
    await filPage.close();

    console.log('\n✅ Both PDFs completed successfully!');
    console.log(`English: /Users/jmbonnevie/Projects/aunt-sallys/manual-en.pdf`);
    console.log(`Filipino: /Users/jmbonnevie/Projects/aunt-sallys/manual-fil.pdf`);
    
    // Verify files exist
    const enPdfExists = fs.existsSync(path.resolve(__dirname, 'manual-en.pdf'));
    const filPdfExists = fs.existsSync(path.resolve(__dirname, 'manual-fil.pdf'));
    
    console.log(`\nFile verification:`);
    console.log(`  manual-en.pdf: ${enPdfExists ? '✓ exists' : '✗ missing'}`);
    console.log(`  manual-fil.pdf: ${filPdfExists ? '✓ exists' : '✗ missing'}`);

  } catch (error) {
    console.error('\n❌ Error during PDF conversion:');
    console.error(error.message);
    if (error.stack) {
      console.error(error.stack);
    }
    process.exit(1);
  } finally {
    if (browser) {
      await browser.close();
    }
  }
})();
